### ISSUE TITLE
build fails when there is a file looks like ".#foo"

### ISSUE THREAD

**Issue by xxks-kkk** [sphinx-doc/sphinx#2368]
I use emacs as my editor and it will generate files start with ".#" (ie. ".#foo"). Then, for some reason, this will make the whole building process fail and generate the error message looks  like this:

`Exception occurred:
  File "/usr/local/lib/python2.7/dist-packages/docutils/io.py", line 238, in __init__
    raise InputError(error.errno, error.strerror, source_path)
InputError: [Errno 2] No such file or directory: u'/home/zeyuan_hu/Documents/zeyuan-s-doc/source/notes/cs/gre/.#verbal.rst'
The full traceback has been saved in /tmp/sphinx-err-kXNrdY.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 1
`

And the problem will be fixed if I manually delete those files. Is there something special with this file format? Thanks!

**Comment by jfbu** [comment]
These files (in my experience) are there only when you have unsaved Emacs buffers: that is when the buffer contents do not match the file contents. Do you really need to launch the build process in such configurations ?

**Comment by xxks-kkk** [comment]
The following is the log file:

```
# Sphinx version: 1.3.5
# Python version: 2.7.3 (CPython)
# Docutils version: 0.12 release
# Jinja2 version: 2.8
# Last messages:
#   Running Sphinx v1.3.5
#   loading pickled environment...
#   done
#   building [mo]: targets for 0 po files that are out of date
#   building [html]: targets for 1 source files that are out of date
#   updating environment:
#   2 added, 1 changed, 0 removed
#   reading sources... [ 33%] notes/cs/gre/.#verbal
# Loaded extensions:
#   sphinx.ext.todo (1.3.5) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/todo.pyc
#   alabaster (0.7.7) from /usr/local/lib/python2.7/dist-packages/alabaster/__init__.pyc
#   sphinx.ext.mathjax (1.3.5) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/mathjax.pyc
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/sphinx/cmdline.py", line 244, in main
    app.build(opts.force_all, filenames)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/application.py", line 266, in build
    self.builder.build_update()
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 251, in build_update
    'out of date' % len(to_build))
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 265, in build
    self.doctreedir, self.app))
  File "/usr/local/lib/python2.7/dist-packages/sphinx/environment.py", line 618, in update
    self._read_serial(docnames, app)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/environment.py", line 638, in _read_serial
    self.read_doc(docname, app)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/environment.py", line 787, in read_doc
    encoding=self.config.source_encoding)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/environment.py", line 147, in __init__
    FileInput.__init__(self, *args, **kwds)
  File "/usr/local/lib/python2.7/dist-packages/docutils/io.py", line 238, in __init__
    raise InputError(error.errno, error.strerror, source_path)
InputError: [Errno 2] No such file or directory: u'/home/zeyuan_hu/Documents/zeyuan-s-doc/source/notes/cs/gre/.#verbal.rst'
```

**Comment by xxks-kkk** [comment]
Well, just curious why those unsaved buffer tmp files can make the build process fail. Sometimes my system crashed with emacs left open and sometimes I forgot to clean up directory before build process and it leads to build process crash. If multiple tmp files exist and I have to run some shell script to clean up and this annoys me.

**Comment by tk0miya** [comment]
It was fixed at #637 once. But it was not perfect.
So I fixed it again at 5df6d09.

Thank you for reporting!

**Comment by xxks-kkk** [comment]
my pleasure :smile:
